hw/Makefile

```diff
@@ -128,14 +128,15 @@ clean-regs-header:
 	rm -r -f ${REG_OUTPUT_SW_DIR}
 
 top-hjson = $(PRJ_DIR)/hw/$*/data/$*.hjson
+version_path = $(PRJ_DIR)/bazel-bin/util/full_version.txt
```

Comment:
Does something guarantee that they gets built?  ie: Can we create a `make` rule or add a `bazelisk.sh build //util:full-version` to the tops_gen/tops_reg rules to make sure this file gets created?

Reply:
Sort of. there is a Bazel rule that builds it: https://cs.opensource.google/opentitan/opentitan/+/master:util/BUILD;drc=8e341cf720794b6d2fceb7ac98505aef6d4876bd;l=20

And that rule is a default dep of the autogened header files: https://cs.opensource.google/opentitan/opentitan/+/master:rules/autogen.bzl;drc=ffdf8e047001c77e4effc7d564906da150ee06bf;l=56

Though it looks like every(?) use of the `autogen_hjson_header` uses the default currently.

In theory though I guess it wouldn't have to.

Reply:
but it would need to be a dep of this makefile somehow as you point out.

Reply:
I see a few option to solve this problem:
1) long term - rewrite this makefile in bazel, and add in dependency invocation of "//util:full_version_file"
2) repurpose existing script "./util/get_workspace_status.sh" and invoke it before topgen.py